File: Backend/Models/Report.cs
namespace Backend.Models;


public class Report
{
    [Key]
    public Guid Id { get; set; } = Guid.NewGuid();

    public required string Title { get; set; }

    [MaxLength(2000)]
    public string? Description { get; set; }

    public DateTimeOffset DateOfPost { get; set; } = DateTimeOffset.UtcNow;

    public User? User { get; set; }

    public List<Comment> Comments { get; set; } = new();

    public List<Media>? Media { get; set; }

    public Region Region { get; set; }

    public Severity Severity { get; set; }

    public ResolutionStatus ResolutionStatus { get; set; }

    public List<Tag> Tags { get; set; } = new List<Tag>();

    public List<User> Followers { get; set; } = new List<User>();
}

File: Frontend/prijavi-ffd/src/interfaces/report.ts

import { User } from './user';
import { Comment } from './comment';
import { Media, Pin } from './media';

export interface Report {
  id: string;
  title: string | null;
  description: string | null;
  dateOfPost: Date;
  username: string | null;
  commentIds: string[] | null;
  mediaIds: string[] | null;
  region: string | null;
  severity: string | null;
  resolutionStatus: string | null;
  pinId: string | null;
  tags: string[] | null;
  followerUsernames: string[] | null;
}

export interface ReportToSend{ 
  title:string; 
  description:string; 
  tagNames:string[]; 
  severityLevel:string; 
  regionName:string; 
  pin?:Pin; }

  